utils/land-patterns.stanza

```diff
@@ -147,11 +147,13 @@ public defn apply-soldermask (mask-shape:Shape):
 public defn apply-soldermask (amount:Double, expand?:True|False):
   val mask-shape = 
     match(pad-shape(self)):
-      (s:Rectangle|Circle|RoundedRectangle|Segment):
+      (s:Rectangle|Circle|RoundedRectangle|Segment|Capsule):
         val func = expand when expand? else offset
         func(s, amount)
-      (s:ChamferedRectangle|DShape|Capsule):
-        expand(s, amount)
+      (s:ChamferedRectangle|DShape):
+        val d = dims(s)
+        val expansion = 1.0 + amount / max(x(d), y(d))
```

Comment:
the logic is as follows: 

`expand(shape, x)` will grow or shrink `shape` by the non-negative factor `x`.

The generator is passed `amount` as an offset, so we need to compute the most sane expansion factor to approximate. This is due to an internal limitation on the implementation of `offset` which does not cover DShape and ChamferedRectangle yet. 

To do this, we find the maximum of the x/y dimensions of the shape. Dividing the amount by this value gives us the expansion as a ratio, adding it to one gives us the overall expansion factor. `amount` may be negative, in which case the shape will shrink.

Reply:
Why this time the expansion parameter has no dimension (ratio of two lengths)? (and for other cases `expand(s, amount)`)

Reply:
Can you clarify what you mean by dimension?

Reply:
unit

Reply:
For `expand (x, factor)`, factor is unitless. It enlarges or shrinks by a scalar. For example:

```
val r1 = Rectangle(1.0, 1.0)
val r2 = expand(r1, 2.0)

r2 == Rectangle(2.0, 2.0)
```

Reply:
The `amount` parameter name is kind of confusing

Reply:
I removed the option to use `expand` and renamed to `expansion-factor` in the code.